src/pages/iou/request/step/IOURequestStepCategory.tsx

```diff
@@ -127,8 +127,8 @@ function IOURequestStepCategory({
 
         IOU.setMoneyRequestCategory(transactionID, updatedCategory);
 
-        if (action === CONST.IOU.ACTION.CATEGORIZE) {
-            Navigation.closeAndNavigate(ROUTES.MONEY_REQUEST_STEP_CONFIRMATION.getRoute(action, iouType, transactionID, report?.reportID ?? '-1'));
+        if (action === CONST.IOU.ACTION.CATEGORIZE && !backTo) {
+            Navigation.navigate(ROUTES.MONEY_REQUEST_STEP_CONFIRMATION.getRoute(action, iouType, transactionID, report?.reportID ?? '-1'));
```

Comment:
I implemented it the same way, but I didn't add this additional check to the if statement, I don't know what it changes